### PR TITLE
udp: do not close socket on bind error on unix

### PR DESCRIPTION
**Pull request by schlamar**
Fixes: #1336

### CODE REVIEW

File: src/unix/udp.c [hunk 1]
```diff
@@ -340,7 +340,6 @@ int uv__udp_bind(uv_udp_t* handle,
   return 0;
 
 out:
-  uv__close(handle->io_watcher.fd);
   handle->io_watcher.fd = -1;
```
Comment (santigimeno) [inline src/unix/udp.c]: this line should also be removed

**Comment by schlamar** [comment]
No CI?! :)

File: src/unix/udp.c (hunk 1 above)
Comment (schlamar) [inline src/unix/udp.c]: Sure :-/

**Comment by santigimeno** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/261/

**Comment by cjihrig** [comment]
This looks like a failure that doesn't happen often:

```
not ok 51 - tcp_write_queue_order
# exit code 134
# Output from process `tcp_write_queue_order`:
# lt-run-tests: src/unix/core.c:166: uv_close: Assertion `0' failed.
```

Not related to UDP, but might be worth running the CI again.
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/263/

**Comment by santigimeno** [comment]
> This looks like a failure that doesn't happen often:

```
not ok 51 - tcp_write_queue_order
# exit code 134
# Output from process `tcp_write_queue_order`:
# lt-run-tests: src/unix/core.c:166: uv_close: Assertion `0' failed.
```

Yeah, it's not related. I've got a fix for it. I'll send it tomorrow.

**Comment by cjihrig** [comment]
Landed in e11dcd4377185359874e67f4962995fdb7e83d19. Thanks!

**Pull request closed without merging**